src/renderer/components/TaskProgress.ts

```diff
@@ -0,0 +1,110 @@
+import { Task, TaskRuntime, TaskHandle } from "@xmcl/task";
+
+import taskProgressTemplate from "../templates/TaskProgress.pug";
+import { ResolvedVersion } from "@xmcl/core";
+
+export default class TaskProgress extends HTMLDivElement {
+	private $progress = () => this.getElementsByClassName("ui progress")[0];
+
+	/*
+	iterable returns tasks in order of insertion.
+	1st task is always the task that is rendered
+	*/
+	private tasks: Map<Task<Object>, TaskRuntime> = new Map();
+
+	private addTask(task: Task<Object>, runtime: TaskRuntime<Task.State>) {
+		this.tasks.set(task, runtime);
+		if (this.tasks.size === 1) {
+			// only 1 task in queue means there were no tasks before
+			$(this).transition("fly up in");
+		}
+	}
+
+	private removeTask(task: Task<Object>) {
+		this.tasks.delete(task);
+		if (this.tasks.size === 0) {
+			// hide progress bar
+			$(this).transition("fly up out");
+		}
+	}
+
+	public constructor() {
+		super();
+	}
+
+	public connectedCallback(): void {
+		if (!this.hasChildNodes())
+			this.render();
+	}
+
+	public render(): void {
+		this.innerHTML = taskProgressTemplate();
+		$(this.$progress()).progress({
+			label: "percent"
+		});
+		this.style.visibility = "hidden";
+	}
+
+	public addInstallTask(task: Task<ResolvedVersion>, instanceName: string): TaskRuntime<Task.State> {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['instanceName' is defined but never used.](https://app.codacy.com/manual/lukechu10/Minecraft-Box-Launcher/pullRequest?prid=5247069)